### PR TITLE
BAEL-1931 Kerberos integration in Java

### PR DESCRIPTION
**Pull request by den314**

### CODE REVIEW

File: spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/client/SampleService.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+@Service
+class SampleService {
+
+	@Value("${app.access-url}")
+	private String endpoint;
+
+	private RestTemplate restTemplate;
+
+	@Autowired
+	public void setRestTemplate(RestTemplate restTemplate) {
```
Comment (pivovarit) [inline spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/client/SampleService.java]: Use constructor-based injection instead? :)

File: spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/kdc/MiniKdcConfigBuilder.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import java.nio.file.Paths;
+import java.util.Objects;
+
+class MiniKdcConfigBuilder {
+
+	private static final int FIELDS_COUNT = 3;
+	private String workDir;
+	private String confDir;
+	private String keytabName;
+	private String[] principals = { "principal/localhost" };
```
Comment (pivovarit) [inline spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/kdc/MiniKdcConfigBuilder.java]: Why not a List? wouldn't it be easier to handle?

File: spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/kdc/MiniKdcConfigBuilder.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+	String[] build() {
+
+		String[] args = new String[FIELDS_COUNT + principals.length];
+
+		args[0] = workDir;
+		args[1] = confDir;
+		args[2] = keytabName;
+
+		for (int i = 0, j = 3; i < principals.length; i++, j++) {
```
Comment (pivovarit) [inline spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/kdc/MiniKdcConfigBuilder.java]: Why not handle that manually? those are just three fields

File: spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/kdc/MiniKdcConfigBuilder.java (hunk 2 above)
Comment (den314) [reply spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/kdc/MiniKdcConfigBuilder.java]: refactored to list usage, thank you

File: spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/kdc/MiniKdcConfigBuilder.java (hunk 3 above)
Comment (den314) [reply spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/kdc/MiniKdcConfigBuilder.java]: using list we get rid of this part

File: spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/client/SampleService.java (hunk 1 above)
Comment (den314) [reply spring-security-sso/spring-security-sso-kerberos/src/main/java/kerberos/client/SampleService.java]: thx - adjusted. Will leave setter for the sake of test showcase.

**Pull request merged**